code/datums/quirks/negative_quirks/blood_deficiency.dm

```diff
@@ -13,10 +13,11 @@
 
 /datum/quirk/blooddeficiency/add(client/client_source)
 	RegisterSignal(quirk_holder, COMSIG_HUMAN_ON_HANDLE_BLOOD, PROC_REF(lose_blood))
-	RegisterSignal(quirk_holder, COMSIG_SPECIES_GAIN, PROC_REF(update_mail))
 
+/datum/quirk/blooddeficiency/add_unique(client/client_source)
 	var/mob/living/carbon/human/human_holder = quirk_holder
 	update_mail(new_species = human_holder.dna.species)
+	RegisterSignal(quirk_holder, COMSIG_SPECIES_GAIN, PROC_REF(update_mail))
```

Comment:
Why in `add_unique`? That's only ever called once. We want the signal on any mob (so `add`), right?